### PR TITLE
Remove dependency check

### PR DESCRIPTION
**Pull request by Natureshadow**
[//]: # (Thanks for helping us out! Cache invalidation sucks, so your help is greatly appreciated!)

## Description

[//]: # (What're you proposing?)

Remove system check for Django version

## Rationale

[//]: # (Why should we implement it?)

Dependency definitions are the correct way to ensure that compatible versions of libraries are used. It is important that maintainers can divert from that, for testing, or if they see fit, or in cases such as a dependency being released and the maintainer not having the time to update the package.
    
Django follows SemVer, so it is quite safe to assume that if 3.1 works, 3.2 and 3.3 will as well.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,19 +16,13 @@ services:
   - postgresql
 
 env:
-  - DJANGO="2.0"
-  - DJANGO="2.1"
```
Comment (Andrew-Chen-Wang) [inline .travis.yml:21]: Approved

File: .travis.yml [hunk 2]
```diff
@@ -4,7 +4,6 @@ dist: xenial
 language: python
 cache: pip
 python:
-  - "3.5"
```
Comment (Andrew-Chen-Wang) [inline .travis.yml:7]: Approved

File: .travis.yml (hunk 1 above)
Comment (Andrew-Chen-Wang) [inline .travis.yml:21]: Approved

File: .travis.yml [hunk 3]
```diff
@@ -17,19 +16,13 @@ services:
   - postgresql
 
 env:
-  - DJANGO="2.0"
-  - DJANGO="2.1"
   - DJANGO="2.2"
   - DJANGO="3.0"
   - DJANGO="3.1"
+  - DJANGO="3.2"
```
Comment (Andrew-Chen-Wang) [inline .travis.yml:22]: The Travis CI is not working properly because I've run out of credits; I'll try to see if I can restart the CI to see if it's working. I'm planning on moving to GitHub Actions once I find the chance to finish off #173

File: CHANGELOG.rst [hunk 4]
```diff
@@ -1,6 +1,11 @@
 What’s new in django-cachalot?
 ==============================
 
+2.3.6
```
Comment (Andrew-Chen-Wang) [inline CHANGELOG.rst:4]: Change minor version to 2.4.0 to signify the dropping of the Django and Python versions. This should also be mentioned in the CHANGELOG.
```suggestion
2.4.0
```

File: README.rst [hunk 5]
```diff
@@ -39,7 +39,10 @@ Table of Contents:
 Quickstart
 ----------
 
-Cachalot officially supports Python 3.5-3.9 and Django 2.0-2.2, 3.0-3.1 with the databases PostgreSQL, SQLite, and MySQL.
+Cachalot officially supports Python 3.6-3.9 and Django 2.2, 3.0, 3.1 and 3.2 with the databases PostgreSQL, SQLite, and MySQL.
```
Comment (Andrew-Chen-Wang) [inline README.rst:42]:
```suggestion
Cachalot officially supports Python 3.6-3.9 and Django 2.2 and 3.0-3.2 with the databases PostgreSQL, SQLite, and MySQL.
```

File: cachalot/__init__.py [hunk 6]
```diff
@@ -1,4 +1,4 @@
-VERSION = (2, 3, 5)
+VERSION = (2, 3, 6)
```
Comment (Andrew-Chen-Wang) [inline cachalot/__init__.py:1]:
```suggestion
VERSION = (2, 4, 0)
```

Comment (Andrew-Chen-Wang) [reply cachalot/__init__.py:1]: From the above comment about dropping version of Python and Django should change minor versions.

File: cachalot/apps.py [hunk 7]
```diff
@@ -10,18 +9,6 @@
     SUPPORTED_ONLY)
 
 
-@register(Tags.compatibility)
-def check_django_version(app_configs, **kwargs):
-    if not (2, 0) <= django_version < (3, 2):
-        return [Error(
```
Comment (Andrew-Chen-Wang) [inline cachalot/apps.py:16]: Replace this with a warning. I agree an error is stupid, but I still highly suggest that at least a warning is mentioned. After every minor version update, Subquery always breaks in cachalot so a warning is still necessary. Ref my comments about async

File: README.rst [hunk 8]
```diff
@@ -39,7 +39,10 @@ Table of Contents:
...
+
+Newer versions of Django 3.x should work as well, but due to cachalot monkey-patching the Django ORM, using a version not
+officially tested and supported might cause unexpected breakage.
```
Comment (Andrew-Chen-Wang) [inline README.rst:45]:
```suggestion
No upper limit is imposed by cachalot. When Django's async ORM or cache is rolled out, there may be huge changes and breakages due to cachalot monkey-patching.
```

With huge monkey patching, I'd be cautious about this. I've had instances of subquery just failing after a Django version minor update. This is either due to the test cases failing to catch certain edge cases, the test suite being incomplete, or usually the ORM is changing somehow.

Additionally, Django 3.3 or 3.4 will introduce the async ORM or the async cache (most likely the cache will come in Django 3.3, but for some reason my PR is within the Django 3.2 updates... oh well) which means monkey patching is going to cause hell for cachalot. Luckily, Andrew hasn't started the async portion as of yet, but the roadmap is saying it'll happen pretty soon.

File: CHANGELOG.rst [hunk 9]
```diff
@@ -1,6 +1,11 @@
...
+-----
+
+- Remove enforced system check for Django version (#175)
```
Comment (Andrew-Chen-Wang) [inline CHANGELOG.rst:7]:
```suggestion
- Remove enforced system check Error for Django version (#175)
- Dropped Django 2.0-2.1
- Dropped Python 3.5
```

**Review by Andrew-Chen-Wang — CHANGES_REQUESTED** [review]
Hey! Sorry for the long hiatus. CI isn't working because no credits. Will try to fix with #173. In the meantime, if there are some weird breakages to Django 3.2, I'll try to update it in my own branch (because my git skills suck). Also not sure if I can edit your PR, but some comments:

**Comment by Andrew-Chen-Wang** [comment]
Rebased and superseded by #181. @Natureshadow your commits included; thanks for the help and patience!
